lib/Sema/CSDiagnostics.cpp

```diff
@@ -5721,6 +5721,10 @@ bool MissingContextualConformanceFailure::diagnoseAsError() {
   auto anchor = getAnchor();
   auto path = getLocator()->getPath();
 
+  if (!path.empty() && path.back().is<LocatorPathElt::OptionalPayload>()) {
```

Comment:
Would it be possible to drop this earlier i.e. while forming a fix for `MissingConformance::forContextual`?

Reply:
Done! Turns out that it was already dropped, it was just not the locator used to record the fixes :)

Reply:
Perfect!